payment-account-age-witness.adoc

```diff
@@ -0,0 +1,196 @@
+
+The victim of the stolen bank account will likely contact his bank and initiate a bank chargeback.
+The Bitcoin seller would in such a case be at risk to lose the received payment in Fiat currency.
+
+We *assume* that the criminal who has access to the bank account is intending to take out the funds of that account as quickly as possible as well as that he intends to do that in few large transactions because with each transaction the risk increases that the fraud gets discovered and the account gets frozen.
+
+With our *trade amount limit* we have already a partial protection against that fraud scheme but we would like to increase the security by having additionally a verification scheme for the *age of the bank account*.
+
+To protect users privacy we use a hashing scheme and only the other trading peer who will receive anyway the payment details during the trade process is able to verify that the provided hash in the offer matches the real account data.
+
+== Overview
+
+The basic idea for that scheme was already discussed on the link:https://forum.bisq.io/t/new-requirement-for-payment-accounts-with-chargeback-risk/2376/65[Bisq Forum [1\]].
+
+This proposal is only relevant for Fiat payment accounts as with Altcoin accounts risks for chargeback does not exist.
+
+We propose an account age of 2 months to be considered safe. It seems very unlikely that a stolen bank account will not get discovered for such a long period. When the user has never traded or if their account age is less than 2 months their trade amount will be limited.
+
+We suggest following limits:
+
+. Account never used in trade: 25% of the default limit (e.g. 0.125 BTC for a default limit of 0.5 BTC)
+. Trade age < 1 months: 50% of the default limit (e.g. 0.25 BTC for a default limit of 0.5 BTC)
+. Trade age 1 - 2 months: 75% of the default limit (e.g. 0.375 BTC for a default limit of 0.5 BTC)
+. Trade age > 2 months: Default limit will be applied (e.g. 0.5 BTC)
+
+_Please note that in future the trade amount limits will be probably derived from the current market price and the target will be likely about 1000 USD._
+
+
+== AgeWitness
+
+When a user makes his *first trade* with a certain Fiat payment account (e.g. SEPA, Zelle,...) he publishes a new object (we call it AgeWitness) to the P2P network.
+
+_Side note:
+The reason why we separate that into a new data structure and don't use the already existing trade statistics data structure is because in future, trade statistic data might get pruned once the historical data gets too large. We don't want to be exposed to the complexity introduced in future pruning solutions nor do we want to add constraints to that pruning. We can assume that the AgeWitness data will never become that large that pruning will become necessary._
```

Comment:
there will be 1 AgeWitness object per user of Bisq. If the ambition is to gain localbitcoins/coinbase userbases, will this scale to such user bases? File is shipped with bisq so maybe not an issue, what's the size of 1M AgeWitness objects, etc. After you thought about it for 1 min just write something on what we could do if it doesn't scale anymore :)

Reply:
There might be multiple per user if he has several fiat accounts. The data only contains a few fields. Have checked the size it's 529 raw bytes (PB size might be smaller) that would be about 2000 entries per MB. If we ship 50 MB it should not be a big issue, so that would support 100 000 entries. We could optimize by using only the hash of the pubKey, the pubKey has 443 bytes so that is the big chunk, which a hash that would be only 32 bytes, so in total: 118 bytes, thus giving us 500 000 entries with 50 MB data. A alternative storage solution could be considered as well once that data gets too big to get shipped with the binary. E.g. loading old signed snapshots from a provider node...

Reply:
I will check i the hash of the pubKey is enough and use that if ok. Otherwise we might reach earlier the limits.